test/format/format.sh

```diff
@@ -467,12 +467,6 @@ resolve()
 		# give the parent recording binary a chance to complete if we are using it
 		[[ ! -z $live_record_binary ]] && sleep 2
 
-		# Check for Sanitizer failures, have to do this prior to success because both can be reported.
```

Comment:
As discussed offline, we should make sure we don't skip an MSan failure by removing this. The call to `report_failure` increments the `failure` counter and format.sh may rely exclusively on this to know when something bad happened.

Reply:
Added test suggeted by @mickgraham-mongodb. Will add him to the review.